standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java

```diff
@@ -316,17 +316,20 @@ private boolean ensureDbInit() {
   }
 
   private boolean runTestQuery() {
+    boolean doTrace = LOG.isDebugEnabled();
     Transaction tx = pm.currentTransaction();
     boolean doCommit = false;
     if (!tx.isActive()) {
       tx.begin();
       doCommit = true;
     }
     // Run a self-test query. If it doesn't work, we will self-disable. What a PITA...
-    String selfTestQuery = "select \"DB_ID\" from " + DBS + "";
+    String selfTestQuery = "select \"DB_ID\" from " + DBS + " WHERE \"DB_ID\"=1";
```

Comment:
this query may or may not have results... I think both is fine... how about changing it to `LIMIT 0`  ? 
then it will not even need a primary-key scan... :)

Reply:
Yes, the results do not matter at this part of the code. Actually I did not want to introduce RDBMS specific clauses like "LIMIT 0" (which is not recognized by Oracle for example), also the "WHERE 1=0" could be a valid "no-results" filter, but again, I do not know if all different DBs like that or not.

Reply:
`where 1=0` sounds good to me...or `DB_ID=-1`; but this will be ok as well